### ISSUE TITLE
Realtime Rollout Blowing up

### ISSUE THREAD

**Issue by charlie-becker** [NCAR/miles-credit#363]
Bug! It appears that the autoregressive rollout feeds the inverse normalization data back into the model. It also seems like the new dynamic forcing is not normalized at all.

  ```
  with torch.no_grad():
        for step in tqdm(range(1, n_steps + 1), desc=f"Rollout {init_str}"):
            y_pred = model(x)

            if flag_mass:
                if step == 1:
                    x_init = x.clone()
                y_pred = opt_mass({"y_pred": y_pred, "x": x_init})["y_pred"]
            if flag_water:
                y_pred = opt_water({"y_pred": y_pred, "x": x})["y_pred"]
            if flag_energy:
                y_pred = opt_energy({"y_pred": y_pred, "x": x})["y_pred"]

            # Inverse-normalize → physical space; squeeze T dim for output.py
            y_phys = (y_pred * denorm_std + denorm_mean).squeeze(2).cpu().numpy()  # (1, C, H, W)

            # Async save via SharedMemory
            shm = SharedMemory(create=True, size=y_phys.nbytes)
            shm_arr = np.ndarray(y_phys.shape, dtype=y_phys.dtype, buffer=shm.buf)
            shm_arr[:] = y_phys
            result = pool.apply_async(
                _save_worker,
                (shm.name, y_phys.shape, y_phys.dtype, init_str, step, fhr_per_step, lat, lon, meta_data, conf),
            )
            results.append(result)

            # Update x for next step (if not last)
            if step < n_steps:
                t_next = init_time + step * dt
                sample_frc = dataset[(t_next, 1)]  # only dynamic_forcing
                batch_frc = _sample_to_batch(sample_frc)
                x_frc, _ = apply_preblocks(preblocks, batch_frc)
                x_frc = x_frc.to(device).float()

                # ERA5Dataset insertion order: [dynfrc | static | prog]
                x[:, :n_dyn, ...] = x_frc
                x[:, static_dim_size:, ...] = torch.from_numpy(y_phys[:, :n_prog, np.newaxis]).to(device)
                # x[:, n_dyn:static_dim_size, ...] — static unchanged
```

**Comment by charlie-becker** [comment]
UPDATE: The preblocks handle the normalization, but the array is overwritten with the inversed normalized data after the preblock instead of beforehand.
